pyexchange/erisx.py

```diff
@@ -0,0 +1,400 @@
+import datetime
+
+from dateutil import parser
+from typing import List
+
+from pyexchange.api import PyexAPI
+from pyexchange.fix import FixEngine
+from pyexchange.model import Order, Trade
+from pymaker.util import http_response_summary
+from pymaker.numeric import Wad
+
+
+class ErisxOrder(Order):
+
+    @staticmethod
+    def from_message(item: dict) -> Order:
+        return Order(order_id=item['oid'],
+                     timestamp=item['created_at'],
+                     pair=item['book'],
+                     is_sell=True if item['side'] == 'sell' else False,
+                     price=Wad.from_number(item['price']),
+                     amount=Wad.from_number(item['amount']))
+
+
+class ErisxTrade(Trade):
+
+    @staticmethod
+    def from_message(trade: dict) -> Trade:
+        return Trade(trade_id=trade['trade_id'],
+                     timestamp=int(parser.isoparse(trade['time']).timestamp()),
+                     pair=trade["contract_symbol"],
+                     is_sell=True if trade['side'] == 'SELL' else False,
+                     price=Wad.from_number(trade['px']),
+                     amount=Wad.from_number(abs(float(trade['qty']))))
+
+
+class ErisxApi(PyexAPI):
+    """
+    Implementation logic for interacting with the ErisX exchange, which uses FIX for order management and
+    market data, and a WebAPI for retrieving account balances.
+
+    ErisX documentation available here: https://www.erisx.com/wp-content/uploads/2020/03/ErisX-FIX-4.4-Spec-V3.3.pdf
+    """
+
+
+    logger = logging.getLogger()
+    timeout = 5
+
+    def __init__(self, fix_trading_endpoint: str, fix_trading_user: str,
+                 fix_marketdata_endpoint: str, fix_marketdata_user: str, password: str,
+                 clearing_url: str, api_key: str, api_secret: str, web_api_only: bool):
+        assert isinstance(fix_trading_endpoint, str)
+        assert isinstance(fix_trading_user, str)
+        assert isinstance(fix_marketdata_endpoint, str)
+        assert isinstance(fix_marketdata_user, str)
+        assert isinstance(password, str)
+
+        assert isinstance(clearing_url, str)
+        assert isinstance(api_key, str)
+        assert isinstance(api_secret, str)
+
+        # enable access from sync_trades and inventory_service without overriding socket
+        if not web_api_only:
+
+            self.fix_trading = ErisxFix(fix_trading_endpoint, fix_trading_user, fix_trading_user, password)
+            self.fix_trading.logon()
+            self.fix_trading_user = fix_marketdata_user
+
+            self.fix_marketdata = ErisxFix(fix_marketdata_endpoint, fix_marketdata_user, fix_trading_user, password)
+            self.fix_marketdata.logon()
+            self.fix_marketdata_user = fix_marketdata_user
+
+        self.clearing_url = clearing_url
+        self.api_secret = api_secret
+        self.api_key = api_key
+
+        # store the account id used to retrieve trades and balances
+        self.account_id = self.get_account()
```

Comment:
This breaks the unit tests, because the mocker isn't available in `setup_method`, so it tries to hit the real URL.  I'm working on it.